### DISCUSSION TITLE
CrazyRadio transmit data limit?

### DISCUSSION THREAD

**Discussion by YWprog** [bitcraze/discussions#1425] [Q&A]
Hi, I was trying to get data from Crazyflie2.1 throught CrazyRadio dongle. I'd like to require the limit of transmitting data through CrazyRadio dongle? 
Here's the situation. My program ran normally with calling 6 parameters, yet It would show "Could not add Stabilizer log config, bad configuration." after I add two more variables to be called. If I kept the amount of called variable in 6, with the added variable mentioned, it could show variable correctly. In addition, CrazyRadio document didn't mention this issue.
Thank for you patience to read this question, I appreciate it!

**Comment by ataffanel** [comment]
The radio link with the Crazyflie works with 32Bytes packet size and the log requires a log block data to fit in one packets. When accounting for required extra data, there is 26 bytes left for the log data payload.

The exact number of variable that can be logged depend of the size of the data. The log subsystem actually has the capability to 'compress' values into a 16 bit floating point value which can provide enough precision for a lot of case. With 4 bytes data, only 6 variables fits in a log packet. More can be fitted with smaller data size.

So the solution when you reach the size limit is to either use smaller data size, the [log example in the lib](https://github.com/bitcraze/crazyflie-lib-python/blob/c09bd869f1a823f26dbc2a3238444e018470a11e/examples/logging/basiclog.py#L81-L82) contains an example of how to compress a value in a FP16, or to split you log into multiple configuration/log block.

The Crazyradio can handle up to about 800 packets per seconds when talking to one Crazyflie (400 when talking to more than one), so at the end that is going to be the limit in amount of data you can transfer from then Crazyflie once you have split the log blocks.
